### PR TITLE
Get the contracts docs up-to-date

### PR DESCRIPTION
**Pull request by compojoom**

### CODE REVIEW

File: contracts/currency_networks_tlbc.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ---
-layout: contracts
+
 title: Currency Networks on Trustlines Blockchain
+sidebar_label: Currency Networks
 ---
 
 **Commit**: [trustlines-protocol/contracts@baebb30](https://github.com/trustlines-protocol/contracts/commit/baebb308d9ef868408c3449250aa9d963cc61d71)
```
Comment (compojoom) [inline contracts/currency_networks_tlbc.md:7]: @cducrest what do we need this for?

Comment (cducrest) [reply contracts/currency_networks_tlbc.md:7]: I don't know, what info?

File: contracts/introduction.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+title: Introduction
+---
+
+The smart contracts are a component of the [Trustlines Protocol](https://trustlines.foundation/protocol.html).
```
Comment (cducrest) [inline contracts/introduction.md:5]:
```suggestion
The smart contracts are components of the [Trustlines Protocol](https://trustlines.foundation/protocol.html).
```

File: contracts/introduction.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+They are currently deployed on the Trustlines Blockchain. 
+Transfers within the Trustlines Network are executed by the smart contracts. All trustlines (i.e. credit lines and balances between users) are notarized on the Trustlines Blockchain. Furthermore, the smart contracts enforce the rules determining how trustlines can be created, used and updated.
```
Comment (cducrest) [inline contracts/introduction.md:7]:
```suggestion
Transfers within currency networks are executed by the smart contracts. All trustlines (i.e. credit lines and balances between users) are notarized on the blockchain. Furthermore, the smart contracts enforce the rules determining how trustlines can be created, used and updated.
```

File: contracts/contributing.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
+---
+id: contributing
+title: Contributing
+---
+
+Welcome and thank you for contributing to the Trustlines Project.
+
+## Get Involved
+
+We appreciate all help! Depending on your knowledge you can help bye
```
Comment (cducrest) [inline contracts/contributing.md:10]:
```suggestion
We appreciate all help! Depending on your knowledge you can help by
```

File: contracts/contributing.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+spreading the word, helping with documentation, reporting issues or even contributing to the source code.
+
+Feel free to checkout the code, play around with it and run the tests.
```
Comment (cducrest) [inline contracts/contributing.md:13]:
```suggestion
Feel free to checkout the code, play around with it, and run the tests.
```

File: contracts/contributing.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+### Request a feature
+Please include a description of what you would like to see implemented and an explanation of why you believe this would
+ be a good addition.
+
+## Open a pull request
+
+### Before starting
+- Follow the Getting started guides to setup your development environment.
+- Ensure that you can run the tests and that they pass. The tests can be run from the root with
+`./pytest tests`
```
Comment (cducrest) [inline contracts/contributing.md:35]:
```suggestion
`pytest tests`
```

File: contracts/currency_networks_tlbc.md (hunk 1 above)
Comment (cducrest) [reply contracts/currency_networks_tlbc.md:7]: I think we should not have this file here

File: contracts/develop/develop.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+---
+title: Start Developing
+---
+To start developing on the smart contracts, you will need the solidity compiler solc version 0.5.8.
```
Comment (cducrest) [inline contracts/develop/develop.md:4]:
```suggestion
To start developing on the smart contracts, you will need the solidity compiler solc version 0.8.0.
```

File: contracts/develop/develop.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+To download it into bin, run:
+
+```bash
+curl -L -o $HOME/bin/solc https://github.com/ethereum/solidity/releases/download/v0.5.8/solc-static-linux && chmod +x $HOME/bin/solc
```
Comment (cducrest) [inline contracts/develop/develop.md:8]:
```suggestion
curl -L -o $HOME/bin/solc https://github.com/ethereum/solidity/releases/download/v0.8.0/solc-static-linux && chmod +x $HOME/bin/solc
```

File: contracts/develop/release.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
+---
+title: Release
+---
+
+For versioning we use [setuptools-scm](https://pypi.org/project/setuptools-scm/). This means the version number is
+derived from git tags. To release a new version of the contracts on PyPI or Docker Hub, simply tag a commit with a valid version
+number either via git, or from [github](https://github.com/trustlines-protocol/contracts/releases/new).
+Make sure to update the changelog accordingly and add all changes since the last released version.
+
+## Pre-commit hooks
```
Comment (cducrest) [inline contracts/develop/release.md:10]: Why is that part in the release category? I think it'd be better in the `Start Developing` section

File: contracts/develop/release.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+You should consider initializing the pre-commit hooks. The
+installed git pre-commit hooks run flake8 and black among other things
+when committing changes to the git repository ::
```
Comment (cducrest) [inline contracts/develop/release.md:14]:
```suggestion
when committing changes to the git repository
```

File: contracts/deploy_tools/usage.md [hunk 11]
```diff
@@ -15,20 +10,20 @@ Usage: tl-deploy [OPTIONS] COMMAND [ARGS]...
...
-  currencynetwork           Deploy a currency network contract.
-  exchange                  Deploy an exchange contract.
-  identity-implementation   Deploy an identity implementation contract.
-  identity-proxy-factory    Deploy an identity proxy factory.
-  test                      Deploy contracts for testing.
+  currencynetwork          Deploy a currency network contract.
+  exchange                 Deploy an exchange contract.
+  identity-implementation  Deploy an identity implementation contract.
+  identity-proxy-factory   Deploy an identity proxy factory.
+  test                     Deploy contracts for testing.
```
Comment (cducrest) [inline contracts/deploy_tools/usage.md:21]: Why's that weird coloured?

Comment (cducrest) [inline contracts/deploy_tools/usage.md:21]:
```suggestion
  Usage: tl-deploy [OPTIONS] COMMAND [ARGS]...

  Commandline tool to deploy the Trustlines contracts

Options:
  --version  Prints the version of the software
  --help     Show this message and exit.

Commands:
  beacon                   Deploy a new beacon contract
  currencynetwork          Deploy a currency network contract.
  deploy-and-migrate       Deploy new currrency networks and migrate old ones
                           to deployed ones.

  exchange                 Deploy an exchange contract.
  identity-implementation  Deploy an identity implementation contract.
  identity-proxy-factory   Deploy an identity proxy factory.
  migration                Migrate old currency networks to new ones.
  test                     Deploy contracts for testing.
  verify-migration         Verify migration from old currency networks to new
                           ones.

```

Comment (cducrest) [inline contracts/deploy_tools/usage.md:21]:
```
Usage: tl-deploy [OPTIONS] COMMAND [ARGS]...

  Commandline tool to deploy the Trustlines contracts

Options:
  --version  Prints the version of the software
  --help     Show this message and exit.

Commands:
  beacon                   Deploy a new beacon contract
  currencynetwork          Deploy a currency network contract.
  deploy-and-migrate       Deploy new currrency networks and migrate old ones
                           to deployed ones.

  exchange                 Deploy an exchange contract.
  identity-implementation  Deploy an identity implementation contract.
  identity-proxy-factory   Deploy an identity proxy factory.
  migration                Migrate old currency networks to new ones.
  test                     Deploy contracts for testing.
  verify-migration         Verify migration from old currency networks to new
                           ones.
```

**Review by cducrest — APPROVED** [review]

File: contracts/develop/release.md (hunk 9 above)
Comment (compojoom) [reply contracts/develop/release.md]: yes, you are right.

File: contracts/deploy_tools/usage.md (hunk 11 above)
Comment (compojoom) [reply contracts/deploy_tools/usage.md:21]: Something with the bash formatter. I removed it.

File: contracts/currency_networks_tlbc.md (hunk 1 above)
Comment (compojoom) [reply contracts/currency_networks_tlbc.md:7]: Leaving it for now there. Will talk with @andfletcher about it.

**Pull request merged**